### PR TITLE
Fix parseHttpHeaderAsNumber to return undefined instead of NaN

### PR DESCRIPTION
**Pull request by jar-stripe**
### Why?

`parseHttpHeaderAsNumber` returns `NaN` for absent, undefined, empty, or non-numeric header values. The return type claims `number` but NaN silently violates that contract. Today this doesn't cause user-visible bugs because the sole callsite (`_getSleepTimeInMS`) has a `Number.isInteger` guard that accidentally rejects NaN — but the safety is coincidental, not intentional.

### What?

- `parseHttpHeaderAsNumber` now returns `number | undefined` instead of `number`
- Returns `undefined` for absent, null, empty, whitespace-only, or non-numeric inputs
- Returns a valid number only for finite numeric strings
- `retryAfter` parameter in `RequestSender` is now optional (`retryAfter?: number`) instead of `number | null`
- Added 14 unit tests covering all edge cases

### See Also

- https://github.com/stripe/stripe-node/issues/2732
- https://jira.corp.stripe.com/browse/RUN_DEVSDK-2458

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -459,9 +459,20 @@ export function parseHttpHeaderAsString<K extends keyof RequestHeaders>(
 
 export function parseHttpHeaderAsNumber<K extends keyof RequestHeaders>(
   header: RequestHeaders[K]
-): number {
-  const number = Array.isArray(header) ? header[0] : header;
-  return Number(number);
+): number | undefined {
```
[the remark this answers is not in the document]
Comment (jar-stripe) [reply src/utils.ts:462]: Good catch — fixed in e35cad9. Parameter type is now `RequestHeaders[K] | null | undefined` to match what the function actually accepts.

**Review by xavdid — APPROVED** [review]
